express-notes-api/server.js

```diff
@@ -0,0 +1,100 @@
+import express from 'express';
+import * as fs from 'node:fs';
+import * as fsPromises from 'node:fs/promises';
+
+const fileData = fs.readFileSync('data.json', 'utf8');
+const data = JSON.parse(fileData);
+const notes = data.notes;
+const writeFileDataJSON = () => fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
+
+const getFunction = (req, res) => {
+  const id = req.params.id;
+  if (id < 1) {
+    res.status(400).send({ error: 'id must be a positive integer' });
```

Comment:
- Make sure your ids are positive _integers_, not just positive _numbers_,
- Make sure `NaN` values are caught by your `id` validation:

---

![image](https://user-images.githubusercontent.com/16143771/226972611-7434676d-63f3-40b8-a38b-c02f84a14155.png)